libs/utils/energy_model.py

```diff
@@ -1152,3 +1153,366 @@ def f(input_row):
             return ret
 
         return inputs.apply(f, axis=1)
+
+###########################################
+# Sanity Checks helpers for Energy Models #
+###########################################
+
+# The utilization rate limit to switch to the overutilization mode
+OVERUTILIZED_RATE = 0.8
+
+def _get_powers_list(node):
+    """
+    Get the list of powers for a given node.
+    :param node: A node of the energy model. It can be a cluster or cpu.
+    :type node: EnergyModelNode
+
+    :returns: The list of powers associates with the given node.
+    """
+    powers = []
+    for freq in node.active_states.keys():
+        powers.append(node.active_states[freq].power)
+    return powers
+
+def _get_capacities_list(node):
+    """
+    Get the list of capacities for a given node.
+    :param node: A node of the energy model.
+    :type node: EnergyModelNode
+
+    :returns: The list of capacities associates with the given node.
+    """
+    cap = []
+    for freq in node.active_states.keys():
+        cap.append(node.active_states[freq].capacity)
+    return cap
+
+def _is_list_strickly_increasing(l):
+    for i in range(len(l)-1):
```

Comment:
For short lists, the fastest measured way is:
sorted(l) == l
This is because sorted is a builtin and avoids part of the bytecode machinery.
You can convince yourself with timeit module.